### PR TITLE
Single RAG example

### PR DESCRIPTION
**Pull request by sumanyu**
Add a simple end-to-end RAG example for our Python SDK

### CODE REVIEW

File: python/README.md [hunk 1]
```diff
@@ -1 +1,62 @@
...
+HAMMING_API_KEY="<YOUR_API_KEY>"
+```
+
+## Running Examples
+
+### Example 1: RAG with single retrieval and single generation
+
+For this example, check out the [source code](./single_retrieval_rag/).
+
+1. Create a dataset using the _hamming_, and copy the _Dataset ID_ that was generated.
```
Comment (mariusbld) [inline python/README.md]: "using the _hamming_ SDK"

File: python/sample_dataset/create.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    Hamming,
+)
+
+from data import sample_data
+
+
+load_dotenv()
+
+HAMMING_API_KEY = os.getenv("HAMMING_API_KEY")
+DATASET_ID = os.getenv("DATASET_ID")
```
Comment (mariusbld) [inline python/sample_dataset/create.py]: we don't need this here, remove

File: python/single_retrieval_rag/run.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                metadata=GenerationParams.Metadata(model="GPT-3"),
+            )
+        )
+
+        return {"output": found_row["aiOutput"]}
+
+    result = hamming.experiments.run(
+        RunOptions(
+            dataset=DATASET_ID,
+            name="Test run python sdk async",
```
Comment (mariusbld) [inline python/single_retrieval_rag/run.py]: maybe we can change the name here a bit?
"Hello world from Python SDK"

**Review by mariusbld — APPROVED** [review]
LGTM, left some comments

File: python/README.md (hunk 1 above)
Comment (sumanyu) [reply python/README.md]: Fixed

File: python/sample_dataset/create.py (hunk 2 above)
Comment (sumanyu) [reply python/sample_dataset/create.py]: Yess; thanks

File: python/single_retrieval_rag/run.py (hunk 3 above)
Comment (sumanyu) [reply python/single_retrieval_rag/run.py]: Yup 100% fixed

**Pull request merged**